src/Intersection.hs

```diff
@@ -0,0 +1,18 @@
+module Intersection where
+
+import Data.Ord
+
+import DataTypes
+
+data Intersection = Intersection {
+  intersectedObject :: Object,
+  distance :: Double
+} deriving Eq
+
+instance Ord Intersection where
+  compare x y
+    | distance x < distance y = LT
+    | distance x > distance y = GT
+    | otherwise = EQ
+  x <= y = distance x <= distance y
```

Comment:
exactly, that's all that's required for the minimal definition. You can define the other functions as well, if you wanted much more efficient versions of them for example, but this is fine for us